### DISCUSSION TITLE
TS0001_power - Total Energy goes up and down

### DISCUSSION THREAD

**Discussion by yannpub** [Koenkk/zigbee2mqtt#24869] [Devices]
I have 2 TuYa devices of this type: [TS0001_power](https://www.zigbee2mqtt.io/devices/TS0001_power.html#tuya-ts0001_power)

After a while, the total energy meter starts to get up and down (looks like it comes back all the time to a kind of max value:
<img width="2491" alt="image" src="https://github.com/user-attachments/assets/65ae16e3-a93d-4881-849e-151773d6619b">

Obviously, that make energy dashboards a bit messy has consumption appears like +20 -20 +50 -50, and it gets worse the more you advance in time.

The only I have found to fix is physically restart the device by cutting the power.

Therefore, here are my open points for discussion:
 --> Has anyone experienced the same behaviour ?
 --> Have you found other ways of having the device report total consumption properly ?
 --> I have seen some alternatives developments here and there for some open source firmwares, but none that seemed ready for general use. Has anyone already flashed other firmwares on these TuYa zigbee devices ?

**Comment by starox** [comment]
Hello, I have one of these module working since serveral month.
I don't have your behaviour, I have a steady linear curve.
It works well.

I can't say for the moment which hardware version I have because of network problems ('No network route' -> fw 20240710 curse)

**Comment by zhvial** [comment]
I have the same problem
It reports correct value and then switches to incorrect. 

![image](https://github.com/user-attachments/assets/43470ff9-186e-4db1-8424-5ee9abe4ff8b)
![image](https://github.com/user-attachments/assets/e7422594-a40d-4ba6-b6bf-1e0696ba0452)

**Reply by zhvial** [reply]
![image](https://github.com/user-attachments/assets/69afffa0-928c-47cd-ba8e-d01c10bed415)
From this chart you can see that total consumption rises when there is power demand from the device.

**Comment by yannpub** [comment]
The behaviour above now occurs with my other device. So, definitely these are to be avoided if needed to be used in the long term.

**Comment by tbazire** [comment]
Hello, 
I too have this kind of problem with this equipment and in addition they flood my zigbee network with requests several times per second.
I still have 3 in cardboard and all of them will end up in the trash, because they are really not good.

**Comment by Pwnyrider** [comment]
Hi,

I got the same issue with this device. First, I saw this strange up and down in the energy chart and after reading the comment from @tbazire I realized that this spamming also happend with my device. 
Re-pair and reboot of the coordinator did not help but removing the power completely was the solution. 
After the device was powered on again, everything went to normal again. 
Spamming stopped as well as the energy consumption sensor stopped this up and down behavior.

My guess is that this is a firmware issue. For now, my workaround is a Home Assistant automation tracking the energy sensor, sending an alarm if the consumption sensor goes "backwards".

**Comment by S474N** [comment]
Same problem with same device. 

I have an electric towel dryer in the bathroom, power 600W. It took me 4 hours today, so the consumption should be about 1.2 kWh. According to HA, the consumption is 22 kWh :D

![image](https://github.com/user-attachments/assets/a44e8220-5b74-46f4-87c2-31f4e3054946)

I think that the problem is that something other than the current consumption in W is used to calculate the energy (by the way, HA is also displayed correctly here, see screenshots).

![image](https://github.com/user-attachments/assets/b576dd59-e6dd-4af6-93e9-fbcdeb8d162e)

![image](https://github.com/user-attachments/assets/1b0b9b3d-51b7-4549-84e2-e064c62bbd83)

`{"current":2.55,"elapsed":6004,"energy":11.33,"last_seen":"2025-06-02T16:36:52.047Z","linkquality":65,"power":584,"power_outage_memory":"off","state":"ON","switch_type":"toggle","update":{"installed_version":-1,"latest_version":-1,"state":null},"voltage":230}`

**Reply by lutz-fechner** [reply]
I installed one of theese today. power changes are popping up in Z2M very late. Never seen such lazy reporting on any other device that supports power metering.

**Comment by LeoeLeoeL** [comment]
I have 5 of these devices and one of them, few days ago started with...
<img width="1630" height="401" alt="image" src="https://github.com/user-attachments/assets/f5ec1704-f7e1-4cbd-bdd6-d378c0ae7a3c" />

I suppose no FW update is expected by Tuya.

**Comment by NoobOnTour024** [comment]
Same issue here:
![Screenshot_20250717_175129_Home Assistant](https://github.com/user-attachments/assets/5f69551c-6f28-42b0-8c2b-191178cdcdc4)

Guess I'll just request a refund.

**Comment by Pwnyrider** [comment]
I found a workaround for this issue. 
When this strange behavior of the device happens do the following:
Stop Z2M, wait a couple seconds and start Z2M again. After that the device behaves normally again. 

Note: 
I tested this workaround maybe 10 times so far. Only once I had to stop and start Z2M three times until it worked. 

Please test it yourself and share your results.

**Comment by S474N** [comment]
@Pwnyrider: so is this a problem at the Z2M level?
<img width="563" height="402" alt="image" src="https://github.com/user-attachments/assets/818dec40-31ad-450e-9096-0896f7b6de13" />

**Reply by Pwnyrider** [reply]
I cannot say this for sure. 
This has to be investigated by a developer who is familiar with the Z2M internals or maybe the firmware installed on the coordinator.

**Reply by NoobOnTour024** [reply]
I'm using the Sonoff zigbee 3.0 dongle E with ember firmware.

**Reply by Pwnyrider** [reply]
I have the Sonoff Zigbee 3.0 dongle P with Z-Stack firmware.

**Comment by LeoeLeoeL** [comment]
It doesn't seem a Z2M issue.
To recover my device I had to powercycle.
Restarting Z2M (even many times) didn't solve.

**Reply by Pwnyrider** [reply]
@LeoeLeoeL  Which coordinator and firmware do you use?

**Reply by LeoeLeoeL** [reply]
Sonoff Dongle E
FW 8.0.2 by Nerivec

**Reply by S474N** [reply]
> @LeoeLeoeL Which coordinator and firmware do you use?

I have SLZB-MR1 - CC2652P7, firmware 20240710

**Comment by RenanAz** [comment]
Having the same exact problem with these, and sadly I bought like 7 of them.
2 installed and gave me problems. One has stopped after I changed the polling to - 1. The other one is still very problematic.
Do we have any fw updates? Any other solution than power cycle?
Using Sonoff 3.0 P 
Thanks
